crates/cheatnet/src/forking/state.rs

```diff
@@ -134,7 +134,14 @@ impl StateReader for ForkStateReader {
                 Ok(value_sf)
             }
             Err(ProviderError::Other(boxed)) => other_provider_error(boxed),
-            Err(ProviderError::StarknetError(StarknetError::ContractNotFound)) => Ok(Default::default()),
+            Err(ProviderError::StarknetError(StarknetError::ContractNotFound)) => {
+                self.cache.borrow_mut().cache_get_storage_at(
+                    contract_address,
+                    key,
+                    Default::default(),
+                );
+                Ok(Default::default())
```

Comment:
nit:
```suggestion
                Ok(Felt::default())
```
Also in the other functions